Python_Backend/dummy_server.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python
+
+import asyncio
+import base64
+import binascii
+import io
+import numpy
+from PIL import Image, UnidentifiedImageError
+import websockets
+import contextvars
+
+import AnswerGenerator
+import DFGenerator
+from StateGenerator import StateGenerator
+import json 
+
+#new imports for dummy server
+import pandas
+import os
+import sys
+
+folder = contextvars.ContextVar('folder', default="")
+iris_itr = contextvars.ContextVar('iris_itr', default=0)
+face_itr = contextvars.ContextVar('face_itr', default=0)
+
+
+# Initiate State Generator with the appropriate models
+facial_state_generator = StateGenerator("../Machine_Learning_Model/smile_neutral_rf.pkl", "FACE")
+iris_state_generator = StateGenerator("../Machine_Learning_Model/iris.pkl", "IRIS")
+
+# Two types of Generators
+facial_answer_generator = AnswerGenerator.FacialAnswerGenerator()
+iris_answer_generator = None  # TODO MAKE THIS THE ACTUAL DATA TYPE
+
+FACE = "FACE"
+IRIS = "IRIS"
+
+# Error Strings
+invalid_state_exception = "ERROR: Invalid State Exception"
+no_face_detected_exception = "ERROR: No Face Detected"
+multi_face_detected_exception = "ERROR: Multiple Faces Detected"
+invalid_model_type = "ERROR: Invalid Model Type"
+
+df_generator_exception = "ERROR: DF Generator Failed"
+state_generator_exception = "ERROR: State Generator Failed"
+answer_generator_exception = "ERROR: Answer Generator Failed"
+
+current_answer = contextvars.ContextVar('current_answer', default=AnswerGenerator.Answer.UNDEFINED)
+
+def get_landmarks(image_data):
+    if (image_data[0] == FACE):
+
+        try: 
+            df = DFGenerator.FacialDFGenerator.generate_df(image_data[1])
+        
+        except DFGenerator.NoFaceDetectedException:
+            return no_face_detected_exception
+        
+        except DFGenerator.MultiFaceDetectedException:
+            return multi_face_detected_exception
+
+        except Exception:
+            return df_generator_exception
+
+
+    elif (image_data[0] == IRIS):
+        
+        try:
+            df = DFGenerator.IrisDFGenerator.generate_df(image_data[1])
+
+        except DFGenerator.NoFaceDetectedException:
+            return no_face_detected_exception
+        
+        except DFGenerator.MultiFaceDetectedException:
+            return multi_face_detected_exception
+
+        except Exception:
+            return df_generator_exception
+
+    else:
+        print("how did this even happen?")
+        return Exception
```

Comment:
Exception Should be raised right? or does the try catch block still catch it?

Reply:
has been modified to raise the exception